### PR TITLE
GLFW: minor fixes for the fences

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: modules/glfw/src/glfwmodule.cpp [hunk 1]
```diff
@@ -75,19 +81,23 @@ void GLFWModule::onProcessorNetworkEvaluationEnd() {
     // its work before we continue. This is needed to make sure that we have textures that are upto
     // data when we render the canvases.
     auto syncObj = glFenceSync(GL_SYNC_GPU_COMMANDS_COMPLETE, 0);
```
Comment (ResearchDaniel) [inline modules/glfw/src/glfwmodule.cpp:83]: Is this not callled when the exception is thrown? Should check if it exist in that case

File: modules/glfw/src/glfwmodule.cpp [hunk 2]
```diff
@@ -75,19 +81,23 @@ void GLFWModule::onProcessorNetworkEvaluationEnd() {
...
-    auto res = glClientWaitSync(syncObj, GL_SYNC_FLUSH_COMMANDS_BIT, 100000000);
+
+    const GLuint64 timeoutInNanoSec = 50'000'000;  // 50ms
+
+    auto res = glClientWaitSync(syncObj, GL_SYNC_FLUSH_COMMANDS_BIT, timeoutInNanoSec);
+    while (res == GL_TIMEOUT_EXPIRED) {
+        res = glClientWaitSync(syncObj, GL_SYNC_FLUSH_COMMANDS_BIT, timeoutInNanoSec);
```
Comment (ResearchDaniel) [inline modules/glfw/src/glfwmodule.cpp:89]: Should there be a max timeout here?

**Review by ResearchDaniel — DISMISSED** [review]
There might be an issue with the destructor

File: modules/glfw/src/glfwmodule.cpp (hunk 2 above)
Comment (petersteneteg) [reply modules/glfw/src/glfwmodule.cpp:89]: maybe, this is more or less what we have in qt, seems to work fine there... usually windows will timeout after 30 sec anyway...

File: modules/glfw/src/glfwmodule.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/glfw/src/glfwmodule.cpp:83]: how do you mean? what exception? you mean if a processor fail in the network?

Comment (petersteneteg) [reply modules/glfw/src/glfwmodule.cpp:83]: this will always be call even if a processor fail, but that should not be a problem

Comment (ResearchDaniel) [reply modules/glfw/src/glfwmodule.cpp:83]: There is an exception thrown in the constructor if not glFenceSync exist. Then it is used here

Comment (petersteneteg) [reply modules/glfw/src/glfwmodule.cpp:83]: if the constructor throws then the destructor is never called

**Comment by petersteneteg** [comment]
@ResearchDaniel poke

**Review by ResearchDaniel — APPROVED** [review]

**Comment by petersteneteg** [comment]
@ResearchDaniel Thanks!

**Pull request merged**